### ISSUE TITLE
Build: work with Accelerate team to get them up and running with Build

### ISSUE THREAD

**Issue by stephenakearns** [Anaconda-Server/anaconda-build#128]

**Comment by damianavila** [comment]
OK, this is the summary of the discussion:

**NOTE**: the emphasis with bold letters is mine, not Graham...

* The current build system is Jenkins: http://jenkins:8080/view/conda%20builds/job/conda_accelerate_pr/
* Jenkins just builds the conda recipe within the repository: https://github.com/ContinuumIO/accelerate/tree/master/condarecipes/accelerate

**Past experience**: 
* The main problem Graham had with the setup before, was that there was no expectation of reliability or uptime, and when things were down or not working as expected, **it was impossible to get assistance** - for non-Linux build queues, they frequently went down or had unexpected behavior, and it could sometimes take several days to get them back up. Eventually he switched to manually checking out the repos on different machines and running builds and tests by hand because it was less time-consuming and didn't mean he had to wait a long time to do a build.
* Additionally, Graham wanted to be able to run a build worker so that he could locally debug issues - the turnaround time for executing a build on an anaconda.org build worker was long, and I was **frequently seeing mysterious failures**, which he thought he could understand much faster if I could locally replicate a similar environment for testing.

**Summary**: 
* It is not so much the current state of any particular issue with Anaconda build that would affect whether it is practical to use Anaconda.org build for Accelerate again, but more whether **they can get responses to issues**, and whether there is an expectation of the build workers being **consistently available**. In the past Graham filed issues on the anaconda-server repo, but **he felt largely ignored** and eventually stopped reporting issues. When build workers were down, it would often take several hours (or more than a day) to get a response on the Anaconda Server channel. 

**Next experience**:
* He'd set up a build that just checked out the repository and tried to build the recipe on each of the supported platforms (Windows, OS X, and Linux) for each Python version (2.7, 3.4 and 3.5) and Numpy version (1.8 with Python 2.7, 1.9 and 1.10 with all supported Pythons) to see if it worked. The only issue with doing this directly is that the build requires access to the mkl-devel package, which currently resides in a channel on filer.corp.continuum.io, but we could add this to an Anaconda.org private channel and then there should be no issues.

**His main question**: If we start using it again and have issues which we report, will we be able to get help with them?

**Comment by damianavila** [comment]
I think this describe well the status and requirement from Accelerate to use the Anaconda build system... it is mostly a matter of support. Discussion can follow up here, but from the sprint point of view, this should be considered DONE. So closing.

**Comment by gmarkall** [comment]
Thanks for writing those notes up, @damianavila !

I just want to clarify a couple of things in the "next experience" section:

* The potential issue that I wrote to you about with the mkl-devel package being only on filer, I resolved already by creating a private channel and uploading it there, so that build workers that have the accelerate channel token can pull down the mkl-devel package from there: https://anaconda.org/accelerate/mkl-devel
* I only set up one build, with linux-64 and Python 3.5, to see whether it would work (and did not set up any more because it failed)... I have not tried to build on all platforms yet. However, the listed combination of platforms (Py2.7/3.4/3.5, Win/OSX/Linux, NP1.8/1.9/1.10) is what we would want to be running builds of if we were using anaconda.org build.
* The build didn't work because the conda-build version in the root environment was too old -1.17.0, which does not recognise the `numpy x.x` syntax in meta.yaml files. I considered changing my build script to include "conda update conda-build -n root" to get around this issue, but I thought it might be obnoxious to other users of the system, as I expected that it would make a lasting change to the build worker (and perhaps other people depend on 1.17.0 for something).
